### ISSUE TITLE
qustion about norm_size in softmax_loss_layer.cu ?

### ISSUE THREAD

**Issue by 0-LL-0** [kuaitoukid/SoftmaxWithLoss-OHEM#3]
norm_size = batch_size_ - hard_size_; 
Thank you for sharing. With hard-size, why add norm-size to loss? So not all the samples are calculated?
I don't understand a bit. Can you explain it?

**Comment by kuaitoukid** [comment]
It is like focal loss which is a soften version of hard mining. If only considering hard examples, the data distribution may not be modeled properly, as a result we add some normal samples to training as a balance

**Comment by 0-LL-0** [comment]
Got it, thank you very much.

**Comment by liuqunzhong** [comment]
@kuaitoukid 你好，请问norm_size = batch_size_ - hard_size_;对这些norm_size 个样本也没有特殊对待啊，这样和普通的softmaxloss有什么区别呢？
int norm_size = batch_size_ - hard_size_;
    if (norm_size > 0) {
      random_shuffle(losses_.begin() + hard_size_, losses_.end());
      for (int i = hard_size_; i < batch_size_; i ++) {
        selected_indexes_.push_back(losses_[i].first);
        top[0]->mutable_cpu_data()[0] += losses_[i].second;
      }
    }
########
假设二分类问题，batchsize=100，hardratio=0.7，hardsize=70，normsize=30，outer_num_=100，inner_num_=1，nthreads = outer_num_ * inner_num_=100.

for (int i = batch_size_; i < nthreads; i ++) {
      ignored_indexes_.push_back(losses_[i].first);
    }
ignored_indexes_里没有东西啊。我刚学习，我是不是算错了？谢谢！！！


谢谢  是我搞错了。参数batch_size_是一个比训练的batch_size小的数值。
